### ISSUE TITLE
Color swatches: new values are no longer saved

### ISSUE THREAD

**Issue by thdoan** [magento/magento2#8861]
### Preconditions

1. Magento v2.1.2
2. Developer mode

### Steps to reproduce

1. Delete default color swatches
2. Add your own colors
3. 

### Expected result

1. You should be able to add as many colors as you want.

### Actual result

1. After awhile (in my case, after 71 colors), new colors are not saved even though you still get the "You saved the product attribute." success message.

Note that this assumes you have done the patch for issue #8826, or else you may not even be able to get this far.

**Comment by thdoan** [comment]
**UPDATE:** I deleted all the colors and re-added them directly in MySQL. Even though all 127 colors are showing up in Admin now, `$attribute->getData('option')` in `processSwatchOptions()` is only retrieving the  first 71 colors, that's why all the colors after the 71st color are not able to be edited.

Is this a bug, or is there a known limit of 71 total colors?

**UPDATE 2:** I  set up a fresh environment and was not able to reproduce this issue. After doing `print_r($optionsArray)` in `prepareOptionIds()` I have narrowed it down to this error, which only occurs in the environment with the issue:

_Fatal error: Uncaught exception 'Exception' with message 'User Error: Some transactions have not been committed or rolled back..._
